src/cart-icon/index.ts

```diff
@@ -0,0 +1,38 @@
+import { alias, tag, Events, Selectors, Tag } from '@storefront/core';
+
+@alias('quantity')
+@tag('gb-cart-icon', require('./index.html'), require('./index.css'))
+class CartIcon {
+  state: CartIcon.State = {
+    quantity: 0
+  };
+
+  constructor() {
+    const cart = this.select(Selectors.cart);
+    this.state = { ...this.state, quantity: cart.content.quantity };
```

Comment:
No need to spread state since it just contains quantity.

Also see the note about const quantity below

Reply:
I don't know why we need to spread state here but that's how we did it in autocomplete

Reply:
It's only needed when you want to preserve other properties but in this case state only contains quantity as a property and you are overwriting it afterwards.